spec/controllers/api/internal/v1/defendants_controller_spec.rb

```diff
@@ -8,39 +8,25 @@
   before do
     authorise_requests!
 
-    stub_common_platform_prosecution_cases_api
-
-    # This call creates the ProsecutionCase and ProsecutionCaseDefendantOffence
-    # in the CDA DB, which are currently required to query the defendant by id
-    # on this api endpoint.
-    # It implies that defendants are not queryable unless their case has been searched
-    # for beforehand, which seems risky (albeit that VCD should always have queried the
-    # case first via its searchinng options).
-    CommonPlatform::Api::SearchProsecutionCase.call(prosecution_case_reference: prosecution_case_reference)
+    allow(Api::Internal::V1::DefendantSerializer)
+      .to receive(:new)
+      .and_raise(Errors::DefendantError, "Too many maat references: [111,222]")
+
+    allow(CommonPlatform::Api::DefendantFinder)
+      .to receive(:call)
+      .and_return(Defendant.new(body: {}))
   end
 
   context "when the CP /prosecutionCases API has `offenceSummary` having more than one `applicationReference` (maat_id)" do
     let(:defendant_id) { "501fd6c4-cc49-4ee9-b74d-4c7b30a792b6" }
 
-    # This case contain a `offenceSummary` having more than one applicationReference
-    let(:prosecution_case_reference) { "44PC1234567" }
-
     describe "GET #show" do
       it "raises HTTP 422 Unprocessable Entity" do
         get :show, params: { id: defendant_id }, as: :json
 
         expect(response).to have_http_status(:unprocessable_entity)
-        expect(response.body).to include("error\":\"Too many maat references")
+        expect(response.body).to include("error\":\"Too many maat references: [111,222]")
```

Comment:
This is only testing that the exception is rescued, which is okay, because it is a controller.
The logic is tested here: https://github.com/ministryofjustice/laa-court-data-adaptor/blob/48ddd38fc2990ea578a077b0fa384751a9aa1576/spec/models/defendant_spec.rb#L144